lib/param_parsing/parameters.go

```diff
@@ -56,19 +56,52 @@ func GetParameters() Params {
 	logger = lib.InitLogger(params.LogLevel)
 	var err error
 	// Read configuration files
-	if tomlFileExists(params) {
-		params, err = getParamsTOML(params)
-	} else if tfSwitchFileExists(params) {
+	// TOML from Homedir
+	if tomlFileExists(params, lib.GetHomeDirectory()) {
+		params, err = getParamsTOML(params, lib.GetHomeDirectory())
+		if err != nil {
+			logger.Fatalf("Failed to obtain settings from TOML config in home directory: %q", err)
+		}
+	}
+
+	// TOML from ChDirPath
+	if tomlFileExists(params, params.ChDirPath) {
```

Comment:
I really don't like this - I did initially remove it and leave to getParamsTOML to take care of dealing with "does the file exist", since they both currently do it (not keen on the split of responsibility).
The _main_ reason I added it back in is that the GetVersionFromTerraformVerison (I think), doesn't have similar checks and if decoding Terraform fails, the entire thing fails.
I'm wondering if all of these methods (GetParamsFrom*) should:
 * Always be run (they can take care of checking if files exist, if applicable)
 * Always fail gracefully (maybe barring the TOML) and maybe just print out a warning if we fail to parse the source.

Mainly because we can't always guarantee that the way we're parsing them is correct (perhaps an old or newer version of the external application) OR the user might have additional templating etc (and be relying on the TOML file).

Reply:
@warrensbox do you have any thoughts on these options? :)

Reply:
@warrensbox did you get a chance to ponder the above? Can this "conversation" be resolved? :)

Reply:
or @yermulnik

Reply:
@MatthewJohn Apologies my inattention, though I can't get what input you're expecting?
I did re-read you comment several times and I still am clueless on what you'd like to hear 🤷🏻

Reply:
Ah sorry - well, if you're happy with how it's currently implemented, then all good :)

Reply:
If it works as intended an as expected, then I'm all good with it unless you think it has to be re-factored or re-worked 😄